virtcontainers/network.go

```diff
@@ -737,6 +803,23 @@ func xconnectVMNetwork(netPair *NetworkInterfacePair, connect bool) error {
 	}
 }
 
+func xsetupVMNetwork(endpoint *VirtualEndpoint) error {
```

Comment:
Why do we need to define this new function here ?
What's the problem that you cannot solve with `xconnectVMNetwork()` ?

Reply:
In https://github.com/kata-containers/runtime/issues/113#issuecomment-402918673. 
xconnectVMNetwork() assumes that a device named endporint.netPair.VirtIface.Name already exists in the shim net ns. 
https://github.com/kata-containers/runtime/blob/ff7b4f6ed7f622217f456a9215f5e7c6215ff916/virtcontainers/network.go#L917-L939
This is OK if we insert a nic named eth0. Because docker creates a veth pair and rename one of it to eth0 and put it in the net ns. But if we insert another nic eth1, what should we do? We are not responsible for the veth creation.

Reply:
I think we can modify function "bridgeNetworkPair", if the veth pair doesn't exists, we skip the bridge creation and don't insert the veth and tap in the bridge(no bridge to insert), then we can reuse `xconnectVMNetwork()` again.

And in future, CNM and CNI will likely call the same CMD interface "kata-runtime update-network", so I think they should share same logic. You can't just split `kata-runtime update-network` functions with current CNM supporting.